### PR TITLE
update pkg/errors && add pump client in vendor

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
add some function to adjust juju/errors in pkg/errors, so update this pkg
add pump client in vendor, I will use pump client to send binlog next pr.

### What is changed and how it works?
```
dep ensure -update github.com/pkg/errors
dep ensure -add github.com/pingcap/tidb-tools/tidb-binlog/pump_client@ 5db58e3b7e6613456551c40d011806a346b2f44a
sh hack/clean_vendor.sh
```

### CODE REVIEW

File: vendor/github.com/pingcap/tidb-tools/pkg/utils/cpu.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package utils
```
Comment (jackysp) [inline vendor/github.com/pingcap/tidb-tools/pkg/utils/cpu.go]: No license?

File: vendor/github.com/pingcap/tidb-tools/pkg/utils/errors.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package utils
```
Comment (jackysp) [inline vendor/github.com/pingcap/tidb-tools/pkg/utils/errors.go]: ditto

File: vendor/github.com/pingcap/tidb-tools/tidb-binlog/node/node.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
+package node
```
Comment (jackysp) [inline vendor/github.com/pingcap/tidb-tools/tidb-binlog/node/node.go]: ditto.

File: vendor/github.com/pingcap/tidb-tools/tidb-binlog/node/registry.go [hunk 4]
```diff
@@ -0,0 +1,174 @@
+package node
```
Comment (jackysp) [inline vendor/github.com/pingcap/tidb-tools/tidb-binlog/node/registry.go]: ditto

File: vendor/github.com/pkg/errors/errors.go [hunk 5]
```diff
@@ -152,7 +152,8 @@ func (f *fundamental) Format(s fmt.State, verb rune) {
 // WithStack annotates err with a stack trace at the point WithStack was called.
 // If err is nil, WithStack returns nil.
 //
-// Deprecated: use AddStack
+// For most use cases this is deprecated and AddStack should be used (which will ensure just one stack trace).
```
Comment (jackysp) [inline vendor/github.com/pkg/errors/errors.go]: PTAL @lysu

File: vendor/github.com/pkg/errors/stack.go [hunk 6]
```diff
@@ -1,10 +1,12 @@
 package errors
```
Comment (jackysp) [inline vendor/github.com/pkg/errors/stack.go]: ditto

File: vendor/github.com/pkg/errors/juju_adaptor.go [hunk 7]
```diff
@@ -2,16 +2,17 @@ package errors
```
Comment (jackysp) [inline vendor/github.com/pkg/errors/juju_adaptor.go]: ditto

**Comment by jackysp** [comment]
I think we should add a license for each file.

**Comment by WangXiangUSTC** [comment]
I add license for file in tidb-tools. 
For pkg/errors, I think we can add license later. @jackysp

File: vendor/github.com/pkg/errors/errors.go (hunk 5 above)
Comment (lysu) [reply vendor/github.com/pkg/errors/errors.go]: it's reviewed in pingcap/errors

**Comment by lysu** [comment]
@jackysp pingcap/errors is ported from pkg/errors, the origin one doesn't have per-file license, but there are a LICENSE file in https://github.com/pingcap/errors/blob/master/LICENSE

**Comment by jackysp** [comment]
OK, I see.

**Comment by july2993** [comment]
LGTM

**Pull request merged**